### PR TITLE
Adding new and missing schema files for Automation

### PR DESCRIPTION
**Pull request by vrdmr**
Contains the following changes:
- Added the missing schema files for resources in the `2015-10-31` stable resources.
- Added the corresponding examples and fixed the existing ones.

### CODE REVIEW

**Comment by vrdmr** [comment]
FYI @finiteattractor

**Comment by vrdmr** [comment]
@Tiano2017 Could you please review this PR and merge it? Thanks.

**Comment by Tiano2017** [comment]
@vrdmr Will take a look.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -211,14 +211,21 @@
...
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/runbooks" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/modules" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/certificates" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/connections" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/variables" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/schedules" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobs" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobSchedules" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: is this going to break existing templates?

File: tests/2017-05-15-preview/Microsoft.Automation.tests.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "tests": [
+    ]
```
Comment (TianoMS) [inline tests/2017-05-15-preview/Microsoft.Automation.tests.json]: add tests here?

**Comment by Tiano2017** [comment]
@vrdmr left a couple of comments. Can you take a look and address them?

File: tests/2017-05-15-preview/Microsoft.Automation.tests.json (hunk 2 above)
Comment (vrdmr) [reply tests/2017-05-15-preview/Microsoft.Automation.tests.json]: The `2017-05-15-preview` version is currently being added and there are no tests for the same right now. I'll work with the team to populate the tests. I'll remove this file and once the tests are in, put the file back in.

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 1 above)
Comment (vrdmr) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: The earlier schema was hand-written and old (2 years ago). I generated the new ones from Autorest and put renamed them here. The resources themselves haven't changed, but I can double check with my team to make sure that this deploymentTemplate does not break existing templates.

Comment (vrdmr) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: Hi @TianoMS,

I tested with existing templates as well as a custom template I made and based on my testing, the changes are backward compatible.

```
varads-mbp:tools varadmeru$ npm test

> template-schema-validation-tools@1.0.0 test /Users/varadmeru/work/microsoft/github/azure-resource-manager-schemas/tools
> mocha runSchemaValidation.js

  validate template files with schema - 
    ✓ running schema validation on 'Template2016.json' (665ms)
    ✓ running schema validation on 'Template2016.preview.json' (265ms)
    ✓ running schema validation on 'TestTemplate.json' (1020ms)
    ✓ running schema validation on 'TestTemplate.preview.json' (807ms)
    ✓ running schema validation on 'Various.json' (289ms)

  5 passing (3s)

```

File: tests/2017-05-15-preview/Microsoft.Automation.tests.json (hunk 2 above)
Comment (vrdmr) [reply tests/2017-05-15-preview/Microsoft.Automation.tests.json]: Added the tests here.

**Comment by vrdmr** [comment]
@Tiano2017 I've fixed the tests and added a test in `tests/2017-05-15-preview/Microsoft.Automation.tests.json`. Could you please take a look.

Thanks,
Varad

**Pull request merged**